Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportInputController.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard.h>
+#include <AzFramework/Input/Devices/Mouse/InputDeviceMouse.h>
+#include <AzToolsFramework/Viewport/ViewportMessages.h>
+#include <AzToolsFramework/Manipulators/ManipulatorManager.h>
+
+#include <EMStudio/AnimViewportInputController.h>
+#include <EMStudio/ManipulatorInteractionRequestBus.h>
+
+
+namespace EMStudio
+{
+    AzToolsFramework::ViewportInteraction::MouseButton AnimViewportInputController::GetMouseButton(
+        const AzFramework::InputChannel& inputChannel)
+    {
+        using AzToolsFramework::ViewportInteraction::MouseButton;
+        using InputButton = AzFramework::InputDeviceMouse::Button;
+        const AzFramework::InputChannelId& id = inputChannel.GetInputChannelId();
+        if (id == InputButton::Left)
+        {
+            return MouseButton::Left;
+        }
+        if (id == InputButton::Middle)
+        {
+            return MouseButton::Middle;
+        }
+        if (id == InputButton::Right)
+        {
+            return MouseButton::Right;
+        }
+        return MouseButton::None;
+    }
```

Comment:
suggestion: I agree with @amzn-jillich it would be great to move these utility functions out of here and into a shared file in `AzToolsFramework`. This code (and some below) is copied from `ViewportManipulatorController.cpp` in `EditorLib`. I understand this is a pain to access right now, but removing this duplication will be super helpful for future use (especially if there happens to be a bug in one or the other).

This _could_ be done as a separate PR but if possible and not too much work I'd include it here. I suggesting making a new file in `AzToolsFramework` called `AzFrameworkToAzToolsFrameworkInputMappings` (or maybe something slightly less long winded 😝) This could be added to the `AzToolsFramework\Viewport\` folder, and then be used by both `EditorLib` and `EMFX`.